packages/neuron-wallet/src/channel/index.ts

```diff
@@ -258,13 +258,21 @@ export class Listeners {
    * @description channel to get wallets
    */
   static checkWalletPassword = () => {
-    return ipcMain.on(Channel.CheckWalletPassword, (e: Electron.Event, { password }: { password: string }) => {
-      e.sender.send(Channel.CheckWalletPassword, {
-        status: ResponseStatus.Success,
-        result: password === '1qaz',
-      })
-      // e.returnValue = password === '1qaz'
-    })
+    return ipcMain.on(
+      Channel.CheckWalletPassword,
+      (e: Electron.Event, { walletID, password }: { walletID: string; password: string }) => {
+        let valid = false
+        wallets().forEach((value: MockWallet) => {
```

Comment:
This doesn't look right. It returns valid when any wallet has the correct password?

Reply:
`if (value.id === walletID && value.password === password)`
No,find the wallet with right id,then check password

Reply:
IMO, there are two steps

1. find the wallet by the id, otherwise return `wallet not found`.
2. verify its password.

Reply:
This loop will not stop even if valid is true.

Reply:
I've refactored this part